File: brd-android/app-core/src/main/java/com/breadwallet/tools/animation/SpringAnimator.java
package com.breadwallet.tools.animation;

import android.content.Context;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.ScaleAnimation;

import com.breadwallet.appcore.R;

/**
 * BreadWallet
 *
 * Created by Mihail Gutan <mihail@breadwallet.com> on 6/24/15.
 * Copyright (c) 2021 Breadwinner AG
 *
 * SPDX-License-Identifier: BUSL-1.1
 */

public class SpringAnimator {
    private static final String TAG = SpringAnimator.class.getName();

    public static void showExpandCameraGuide(final View view) {
        if (view != null) {
            view.setVisibility(View.GONE);
        }
        ScaleAnimation trans = new ScaleAnimation(0.0f, 1f, 0.0f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        trans.setDuration(800);
        trans.setInterpolator(new DecelerateOvershootInterpolator(1.5f, 2.5f));
        if (view != null) {
            view.setVisibility(View.VISIBLE);
            view.startAnimation(trans);
        }

    }

    /**
     * Shows the springy animation on views
     */
    public static void springView(final View view) {
        if (view == null) return;
        ScaleAnimation trans = new ScaleAnimation(0.8f, 1f, 0.8f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        trans.setDuration(1000);
        trans.setInterpolator(new DecelerateOvershootInterpolator(0.5f, 1f));
        view.setVisibility(View.VISIBLE);
        view.startAnimation(trans);

    }
    /**
     * Shows the springy animation on views
     */
    public static void shortSpringView(final View view) {
        if (view == null) return;
        ScaleAnimation trans = new ScaleAnimation(0.9f, 1f, 0.9f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        trans.setDuration(200);
        trans.setInterpolator(new DecelerateOvershootInterpolator(1.3f, 1.4f));
        view.setVisibility(View.VISIBLE);
        view.startAnimation(trans);

    }

    /**
     * Shows the springy bubble animation on views
     */
    public static void showBubbleAnimation(final View view) {
        if (view == null) return;
        ScaleAnimation trans = new ScaleAnimation(0.75f, 1f, 0.75f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        trans.setDuration(300);
        trans.setInterpolator(new DecelerateOvershootInterpolator(1.0f, 1.85f));
        view.setVisibility(View.VISIBLE);
        view.startAnimation(trans);
    }

    public static void failShakeAnimation(Context context, View view) {
        if (view == null) return;
        Animation shake = AnimationUtils.loadAnimation(context, R.anim.shake);
        view.setVisibility(View.VISIBLE);
        view.startAnimation(shake);
    }

}

File: brd-android/app-core/src/main/java/com/breadwallet/tools/animation/SpringAnimator.kt
/*
 * BreadWallet
 *
 * Created by Amit Goel <amit.goel@breadwallet.com> on 10/08/2021
 * Copyright (c) 2021 Breadwinner AG
 *
 * SPDX-License-Identifier: BUSL-1.1
 */

/**
 * BreadWallet
 *
 * Created by Mihail Gutan <mihail></mihail>@breadwallet.com> on 6/24/15.
 * Copyright (c) 2021 Breadwinner AG
 *
 * SPDX-License-Identifier: BUSL-1.1
 */
package com.breadwallet.tools.animation

import android.content.Context
import android.view.View
import android.view.animation.ScaleAnimation
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import com.breadwallet.appcore.R

object SpringAnimator {
    fun showExpandCameraGuide(view: View?) {
        if (view != null) {
            view.visibility = View.GONE
        }
        val trans = ScaleAnimation(
            0.0f, 1f, 0.0f, 1f,
            Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        trans.duration = 800
        trans.interpolator = DecelerateOvershootInterpolator(1.5f, 2.5f)
        if (view != null) {
            view.visibility = View.VISIBLE
            view.startAnimation(trans)
        }
    }

    /**
     * Shows the springy animation on views
     */
    fun springView(view: View?) {
        if (view == null) return
        val trans = ScaleAnimation(
            0.8f, 1f, 0.8f, 1f,
            Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        trans.duration = 1000
        trans.interpolator = DecelerateOvershootInterpolator(0.5f, 1f)
        view.visibility = View.VISIBLE
        view.startAnimation(trans)
    }

    /**
     * Shows the springy animation on views
     */
    fun shortSpringView(view: View?) {
        if (view == null) return
        val trans = ScaleAnimation(
            0.9f, 1f, 0.9f, 1f,
            Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        trans.duration = 200
        trans.interpolator = DecelerateOvershootInterpolator(1.3f, 1.4f)
        view.visibility = View.VISIBLE
        view.startAnimation(trans)
    }

    /**
     * Shows the springy bubble animation on views
     */
    fun showBubbleAnimation(view: View?) {
        if (view == null) return
        val trans = ScaleAnimation(
            0.75f, 1f, 0.75f, 1f,
            Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f
        )
        trans.duration = 300
        trans.interpolator = DecelerateOvershootInterpolator(1.0f, 1.85f)
        view.visibility = View.VISIBLE
        view.startAnimation(trans)
    }

    fun failShakeAnimation(context: Context?, view: View?) {
        if (view == null) return
        val shake = AnimationUtils.loadAnimation(context, R.anim.shake)
        view.visibility = View.VISIBLE
        view.startAnimation(shake)
    }
}